src/datasets/packaged_modules/audiofolder/audiofolder.py

```diff
@@ -0,0 +1,161 @@
+import glob
+import os
+from typing import List
+
+import datasets
+from datasets.tasks import AutomaticSpeechRecognition
+
+
+class AudioFolderConfig(datasets.BuilderConfig):
+    """BuilderConfig for AudioFolder."""
+
+    def __init__(self, *args, sampling_rate=None, transcripts_filename="transcripts.txt", **kwargs):
+        if not sampling_rate:
+            raise ValueError(
+                "To load an audio dataset, you must provide the `sampling_rate` parameter. "
+                "For example: "
+                '>>> load_dataset("audiofolder", data_dir="path/to/dir", sampling_rate=16_000)'
+            )
+        self.sampling_rate = sampling_rate
+        self.transcripts_filename = transcripts_filename
+
+        super(AudioFolderConfig, self).__init__(*args, **kwargs)
+
+
+class AudioFolder(datasets.GeneratorBasedBuilder):
+    BUILDER_CONFIG_CLASS = AudioFolderConfig
+
+    AUDIO_EXTENSIONS: List[str] = []  # definition at the bottom of the script
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            features=datasets.Features(
+                {
+                    "path": datasets.Value("string"),
+                    "audio": datasets.Audio(sampling_rate=self.config.sampling_rate),
+                    "text": datasets.Value("string"),
+                }
+            ),
+            task_templates=[AutomaticSpeechRecognition(audio_file_path_column="path", transcription_column="text")],
```

Comment:
you can update things here:
```suggestion
            task_templates=[AutomaticSpeechRecognition(audio_column="audio", transcription_column="text")],
```